homeassistant/components/arwn/sensor.py

```diff
@@ -30,26 +30,30 @@ def discover_sensors(topic, payload):
             unit = TEMP_FAHRENHEIT
         else:
             unit = TEMP_CELSIUS
-        return ArwnSensor(name, "temp", unit)
+        return ArwnSensor(topic, name, "temp", unit)
     if domain == "moisture":
         name = f"{parts[2]} Moisture"
-        return ArwnSensor(name, "moisture", unit, "mdi:water-percent")
+        return ArwnSensor(topic, name, "moisture", unit, "mdi:water-percent")
     if domain == "rain":
         if len(parts) >= 3 and parts[2] == "today":
             return ArwnSensor(
-                "Rain Since Midnight", "since_midnight", "in", "mdi:water"
+                topic, "Rain Since Midnight", "since_midnight", "in", "mdi:water"
             )
         return (
-            ArwnSensor("Total Rainfall", "total", unit, "mdi:water"),
-            ArwnSensor("Rainfall Rate", "rate", unit, "mdi:water"),
+            ArwnSensor(topic + "/total", "Total Rainfall", "total", unit, "mdi:water"),
+            ArwnSensor(topic + "/rate", "Rainfall Rate", "rate", unit, "mdi:water"),
         )
     if domain == "barometer":
-        return ArwnSensor("Barometer", "pressure", unit, "mdi:thermometer-lines")
+        return ArwnSensor(topic, "Barometer", "pressure", unit, "mdi:thermometer-lines")
```

Comment:
This is same value as when domain is `temperature`?

Reply:
It's not the same value. The topic is an mqtt full path topic, unique device.